### ISSUE TITLE
This sounds different than the original, some notes are missed or off key.

### ISSUE THREAD

**Issue by jscuster** [JamesNewton/BitShift-Variations-unrolled#5]
So I have the original and this version. I'm on Windows, because reasons, so I compiled them both with "tcc <code.c".

If you compare the output of both, the unrolled version is different. The code seems to match the original, so I don't understand why.

When the song starts, the third, fifth, and seventh notes, for example, are different. When the high notes come in, the ninth note is also off. I didn't listen through and pick all the bad notes out, but hopefully this gives someone an idea. Though all the notes I wrote here were odd, I'm not sure that holds true past the first few seconds. For example, the bass notes 3, 5, and 7 are bad, but 1 and 9 seem ok. In the high section, the first 8 notes are fine, the 9th is off. I'm scratching my head on this one.

Also, the unrolled version doesn't match the Computerphile video, but my original does, so I think my compilation and original source are good.

Also also, the main function is missing a } at the bottom.

If I come up with a fix, I'll fork and do a pull request, but thought better heads than mine might have some ideas too.

Thanks.

**Comment by JamesNewton** [comment]
Thanks for checking on this!

I think the order of operations is unrolled incorrectly in the "3&x&(i*((3&i>>16?" part?

**Comment by jscuster** [comment]
Thanks for the hint, turned out to be something else. I submitted a pull request.

The problem was that the conditional that sets the value of c was:

if (b == 0) {

It should be:
if (b != 0) {

Seems like just a typo, the explanation at the top of the code shows it correctly.

I also added the missing } at the bottom of the main function.

Because my poor little brain got stuck for a bit, I added a return statement at the bottom of g, I thought it would make it more clear that the function returns b. Feel free to remove it, maybe I missed something.

Thank you for reading, I hope this proves helpful.
